content/what-is-cross-origin-resource-sharing/index.md

```diff
@@ -0,0 +1,166 @@
+Host: http://api-emailpassword.demo.supertokens.com
+Access-Control-Request-Method: GET
+Access-Control-Request-Headers: content-type,fdi-version,rid
+Origin: https://emailpassword.demo.supertokens.com
+```
+
+So that's our pre-flight request. Breaking this down, we have four key things to look at.
+
+-   `Host` - The "host" of the resource that we're requesting. For us, that's `api-emailpassword.demo.supertokens.com`
+-   `Access-Control-Request-Method` - The method of the request being made by our operation. This can be any of the HTTP request methods, including `GET`, `POST`, `PUT`, `DELETE`, and `CONNECT`.
+-   `Access-Control-Request-Headers` - A comma-separated list of [HTTP headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers) that would be used in the actual request.
+-   `Origin` - Where the request is coming from. For us, that's our previous domain - `emailpassword.demo.supertokens.com`
+
+After getting this pre-flight / `OPTIONS` request, the API server sends over a pre-flight response. Here's our response from `api-emailpassword.demo.supertokens.com`.
+
+```
+HTTP/1.1 204 No Content
+Access-Control-Allow-Origin: https://emailpassword.demo.supertokens.com/
+Access-Control-Allow-Credentials: true
+Access-Control-Allow-Methods: GET,PUT,POST,DELETE
+Access-Control-Allow-Headers: content-type,rid,fdi-version,anti-csrf
+```
+
+Let's break this response down.
+
+-   `Access-Control-Allow-Origin` - The origins that the API server has whitelisted
+-   `Access-Control-Allow-Credentials` - The server telling us whether the actual request can include cookies in it, or that the response of the actual request can set-cookies. In our case, cookies refer to the session tokens of the user, which act as the credentials of the user once they're signed in.
+-   `Access-Control-Allow-Methods` - A comma-separated list of HTTP methods that the API domain allows for cross-origin requests
+-   `Access-Control-Allow-Headers` - A comma-separated list of HTTP headers that the API domain allows for cross-origin requests
+
+The browser then takes this response from the API server to determine if the actual request should be sent. If the response from the API doesn't include the requested origin, methods, or headers from the preflight request, then the browser will not send the actual request.
+
+## CORS actual request
+
+If the response from the API includes the requested origin, it's time to send the actual `POST` request to sign in.
+
+```
+POST /auth/signin HTTP/1.1
+Host: http://api-emailpassword.demo.supertokens.com/content-type: application/json
+fdi-version: 1.15
+rid: emailpassword
+Content-Length: 92
+Origin: https://emailpassword.demo.supertokens.com/
+```
+
+Note that this request includes both the `Host` and `Origin` headers. In addition, we see the content-type header show up in the request (near the end of the `Host` request line). Looking back to the pre-flight response, we can see that content-type is one of the allowed headers.
```

Comment:
```suggestion
Note that the origin and the headers that are sent (`fdi-version`, `rid`, `content-type`), are whitelisted by the server, and the browser knows this because of the pre-flight response.
```